### PR TITLE
diag(bitnet): compare server probe logits

### PR DESCRIPTION
**Pull request by EffortlessSteven**
## Summary
- report Rust selected logits for reference-server probability probe IDs
- surface candidate logit gaps under `probability_probe_candidate_logits` for CPU and strict A770 comparisons
- keep reference-server probability probes diagnostic-only and explicitly not reference token IDs or raw logits

## Validation
- `cargo test --locked -p xtask --no-default-features --bin xtask bitnet_reference_compare -- --nocapture`
- `cargo check --locked -p xtask --no-default-features`
- `rustfmt --check --edition 2024 --config skip_children=true xtask\src\bitnet_reference_compare.rs`
- `git diff --check`
- real diagnostics:
  - expanded Rust CPU first-token run with server-probe `--logit-id` values
  - expanded strict A770 first-token run with server-probe `--logit-id` values
  - `cargo run --locked -p xtask --no-default-features -- bitnet-reference-compare --reference target/a770-diagnostic/bitnet-reference-run-gguf-tokenizer-fix.json --reference-server target/a770-diagnostic/bitnet-reference-server-run-first-token.json --cpu target/a770-diagnostic/reference-plan-cpu-first-token.json --a770 target/a770-diagnostic/reference-plan-a770-first-token.json --output target/a770-diagnostic/bitnet-reference-compare-with-server-first-token.json --format json`

## Diagnostic result
- CPU and strict A770 still select `.ps` / token `58428`
- reference server selected content remains `2`
- server-probability probe token `17` is the best Rust candidate, but remains about 8.7 logits below Rust argmax

## Claim boundary
- diagnostic only
- does not claim reference generated token IDs
- does not claim reference top logits
- does not promote Rust/reference parity
- does not promote A770 semantic quality, selected attention, resident KV, full support residency, full device residency, or completion

### CODE REVIEW

**Comment by EffortlessSteven** [comment]
Closing as superseded by the current A770 diagnostic branch-chain and ledgered lineage. The active diagnostic path has advanced through #5720, #5722, and #5725; this PR is retained as historical diagnostic evidence only and is not a current merge candidate. No claim promotion is implied.
